library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaSubtitle.java

```diff
@@ -28,45 +28,44 @@
  */
 /* package */ final class SsaSubtitle implements Subtitle {
 
-  private final Cue[] cues;
-  private final long[] cueTimesUs;
+  private final List<List<Cue>> cues;
+  private final List<Long> cueTimesUs;
 
   /**
    * @param cues The cues in the subtitle.
    * @param cueTimesUs The cue times, in microseconds.
    */
-  public SsaSubtitle(Cue[] cues, long[] cueTimesUs) {
+  public SsaSubtitle(List<List<Cue>> cues, List<Long> cueTimesUs) {
     this.cues = cues;
     this.cueTimesUs = cueTimesUs;
   }
 
   @Override
   public int getNextEventTimeIndex(long timeUs) {
     int index = Util.binarySearchCeil(cueTimesUs, timeUs, false, false);
-    return index < cueTimesUs.length ? index : C.INDEX_UNSET;
+    return index < cueTimesUs.size() ? index : C.INDEX_UNSET;
   }
 
   @Override
   public int getEventTimeCount() {
-    return cueTimesUs.length;
+    return cueTimesUs.size();
   }
 
   @Override
   public long getEventTime(int index) {
     Assertions.checkArgument(index >= 0);
-    Assertions.checkArgument(index < cueTimesUs.length);
-    return cueTimesUs[index];
+    Assertions.checkArgument(index < cueTimesUs.size());
+    return cueTimesUs.get(index);
   }
 
   @Override
   public List<Cue> getCues(long timeUs) {
     int index = Util.binarySearchFloor(cueTimesUs, timeUs, true, false);
-    if (index == -1 || cues[index] == Cue.EMPTY) {
+    if (index == -1 || cues.get(index).isEmpty()) {
```

Comment:
I think you can get rid of the empty check, since we'll just return **the** empty list below anyway (right?)

Reply:
Yes, that's right.